### DISCUSSION TITLE
Odd isDirty() wasChanged() results in post-save hooks

### DISCUSSION THREAD

**Discussion by peanutsmcgee** [laravel/framework#40182] [Q&A]
I've noticed some odd behavior with `isDirty` and `wasChanged` when used within a post-save event hook such as `saved` `updated` and `created`. Both are evaluating to true, but shouldn't the `isDirty` be false?

This is the code I tested it with:
```
$model = Model::first();
$model->name = 'new name';
$model->save();

self::saved(function ($model) {
	dump( 'isDirty: ' $model->isDirty('name'));
	dump( 'wasChanged: ' $model->wasChanged('name'));
});
```

Output:
```
isDirty: true
wasChanged: true
```

The way I understand things, `isDirty` is to be used before a model is saved, and `wasChanged` is to be used after a model is saved. Wouldn't this example be considered "after the model was saved"? Is this a bug, or are these post-save hooks supposed to act this way?

**Comment by irfanahmad7272** [comment]
I am facing same issues on live environment, I am using these methods inside a job, executed by horizon at live, but it works wrongly. Only at local environment when we run job via queue work, it works find. Not sure, what can be the issue but couldn't find any proper answer for this.

**Comment by cosmastech** [comment]
Put up a PR for this. https://github.com/laravel/framework/pull/47325

**Comment by cosmastech** [comment]
@peanutsmcgee the PR was rejected as a BC.
